qualtran/bloqs/util_bloqs_test.py

```diff
@@ -55,6 +57,34 @@ def test_util_bloqs():
     assert no_return is None
 
 
+@frozen
+class TestPartition(Bloq):
+    test_bloq: Bloq
+
+    @cached_property
+    def bitsize(self):
+        return sum(reg.total_bits() for reg in self.test_bloq.signature)
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature.build(test_regs=self.bitsize)
+
+    def build_composite_bloq(self, bb: 'BloqBuilder', test_regs: 'SoquetT') -> Dict[str, 'Soquet']:
+        bloq_regs = self.test_bloq.signature
+        partition = Partition(self.bitsize, bloq_regs)
+        out_regs = bb.add(partition, x=test_regs)
+        out_regs = bb.add(self.test_bloq, **{reg.name: sp for reg, sp in zip(bloq_regs, out_regs)})
+        test_regs = bb.add(
+            partition.dagger(), **{reg.name: sp for reg, sp in zip(bloq_regs, out_regs)}
+        )
+        return {'test_regs': test_regs}
+
+
+def test_partition():
+    bloq = TestPartition(test_bloq=TestCNOT())
+    assert_valid_bloq_decomposition(bloq)
```

Comment:
Can we add a test that verifies that `CompositeBloq` using `Partition` works fine when converted to a Cirq circuit?

Reply:
I tested through conversion to a unitary, is this enough?